apps/pixeloven/site/docs/recommendations/react-components.mdx

```diff
@@ -75,4 +75,53 @@ ADD LINK TO DOCUMENTATION ON ROUTES
 
 ## Component composition
 
-ADD DOCUMENTATION ON COMPONENT COMPOSITION
+When it comes to extracting components, there are two important factors to consider:
+1. complexity of code (e.g. conditional rendering and other business logic)
+2. reusability of code
+
+We use the atomic design as our philosophy for creating reusable React components, and create a new storybook entry for each new atomic design component. However, sometimes an atomic design component itself might include complex business logic for rendering, or code that is reused only within that component, and extracting some of its pieces into their own components would help improve readability and maintainability of the code. If you see that these pieces need to be extracted into their own component, yet they do not warrant creating a new atomic design component in storybook, then simply create new React component files directly within the same storybook folder.
+
+In the example below, `CampaignMetadataIndicator`, is a molecule storybook component and it requires complex business for rendering several indicators. The `VelocityIndicator` and `ProximityIndicator` in the end both render the same molecule component, however they use widely different business logic and copy. Extracting these implementations into two separate components lets us clearly see the logic that the parent, `CampaignMetadataIndicator`, uses to decide which indicator to render. `VelocityIndicator` and `ProximityIndicator`, in turn, are responsible for logic specific to them. Since they are not being used anywhere else on the site, they do not warrant creation of a new storybook component (e.g. atom, molecule, ...), so they are kept within the same directory as their parent, `molecules/CampaignMetadataIndicator`.
+
+In this case, both complexity and reusability played a role in the decision to extract the components. Reusability is frequently a more obvious reason, however, complexity alone could warrant extraction, depending on the situation. The goal is to improve readability of the code for us humans, which will make it easier for us to reason about and modify the code as needed in the future.
+
+As [React docs](https://reactjs.org/docs/components-and-props.html#extracting-components) put it: "Don’t be afraid to split components into smaller components." :)
+
+```
+src/shared/components/molecules/CampaignMetadataIndicator
```

Comment:
We should probably generalize this example a bit more so it's not so specific to an employer.

Reply:
Updated - [commit](https://github.com/pixeloven/pixeloven/pull/288/commits/adf3cc50d999d68edbecc482a28605c541249218)